runtime/tywrap_bridge_core.py

```
@@ -849,18 +867,168 @@ def serialize(obj, *, force_json_markers, torch_allow_copy=False):
+                item_key, item = next(iterator)
+            except StopIteration:
+                active_ids.remove(id(current))
+                parent[key] = output
+                continue
+            stack.append(frame)
+            if not (isinstance(item_key, (str, int, float, bool)) or item_key is None):
+                invalid_path = _invalid_key_path(path, item_key)
+                raise TypeError(
+                    f'keys must be str, int, float, bool or None, not '
+                    f'{type(item_key).__name__} at {invalid_path}'
+                )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)